### PR TITLE
Core to ansyn - part B

### PR DESCRIPTION
**Pull request by yairtawil**

### CODE REVIEW

File: src/app/@ansyn/ansyn/modules/plugins/openlayers/mapSourceProviders/open-layers.map-source-provider.ts [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+	create(metaData: ICaseMapState): Promise<any[]> {
 		const source = this.getXYZSource(metaData.data.overlay.imageUrl);
 		const extent = this.getExtent(metaData.data.overlay.footprint);
 		const tileLayer = this.getTileLayer(source, extent);
-		return [tileLayer];
+		this.setExtraData(tileLayer, { [ImageryLayerProperties.FOOTPRINT]: metaData.data.overlay.footprint });
+		return Promise.resolve(tileLayer);
+	}
+
+	generateLayerId(metaData: ICaseMapState) {
```
Comment (asafMasa) [inline src/app/@ansyn/ansyn/modules/plugins/openlayers/mapSourceProviders/open-layers.map-source-provider.ts]: in the future - open-layer plugin will be separated from ansyn package

File: src/app/@ansyn/imagery/public_api.ts [hunk 2]
```diff
@@ -1,29 +1,3 @@
-export { IOverlaysCriteriaOptions } from './model/overlay.model';
-
-export { GeoRegisteration } from './model/overlay.model';
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/public_api.ts]: 😍

File: src/app/@ansyn/imagery/public_api.ts [hunk 3]
```diff
@@ -1,29 +1,3 @@
...
-export { IContextEntity } from './model/case.model';
-export { ICaseBooleanFilterMetadata, ICaseState, IOverlaysManualProcessArgs } from './model/case.model';
-export { ICaseMapsState, IDilutedCaseState, ImageManualProcessArgs } from './model/case.model';
-export { IWorldViewMapState } from './model/case.model';
-export { CaseEnumFilterMetadata } from './model/case.model';
-export { ICaseMapState } from './model/case.model';
-export { IOverlay, IDilutedOverlay, Overlay, IOverlaysCriteria } from './model/overlay.model';
-
 export { MAP_SOURCE_PROVIDERS_CONFIG, IMapSourceProvidersConfig } from './model/base-map-source-provider';
 export { ICaseMapPosition, CaseMapExtent, CaseMapExtentPolygon, ICaseMapProjectedState } from './model/case-map-position.model';
```
Comment (asafMasa) [inline src/app/@ansyn/imagery/public_api.ts]: rename ICaseMapPosition, CaseMapExtent, CaseMapExtentPolygon, ICaseMapProjectedState
remove the "Case" word

File: src/app/@ansyn/map-facade/actions/imagery-status.actions.ts [hunk 4]
```diff
@@ -21,29 +19,29 @@ export enum ImageryStatusActionTypes {
 export class ToggleFavoriteAction implements Action {
 	type: string = ImageryStatusActionTypes.TOGGLE_OVERLAY_FAVORITE;
 
-	constructor(public payload: { id: string, value: boolean, overlay?: IOverlay }) {
+	constructor(public payload: { id: string, value: boolean, overlay?: any }) {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/actions/imagery-status.actions.ts]: don't like it
add TODO: to move those actions elsewhere

File: src/app/@ansyn/map-facade/actions/map.actions.ts [hunk 5]
```diff
@@ -74,7 +78,7 @@ export const MapActionTypes = {
 
 export interface IContextMenuShowPayload {
 	point: Point;
-	overlays: IOverlay[];
+	overlays: any[];
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/actions/map.actions.ts]: add TODO to move elsewhere

File: src/app/@ansyn/map-facade/actions/map.actions.ts [hunk 6]
```diff
@@ -97,7 +101,7 @@ export class PointToRealNorthAction implements Action {
 export class PositionChangedAction implements Action {
 	type = MapActionTypes.POSITION_CHANGED;
 
-	constructor(public payload: { id: string, position: ICaseMapPosition, mapInstance: ICaseMapState }) {
+	constructor(public payload: { id: string, position: ICaseMapPosition, mapInstance: any }) {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/actions/map.actions.ts]: mapInstance -> add TODO

File: src/app/@ansyn/map-facade/alerts/alert-component.directive.ts [hunk 7]
```diff
@@ -1,12 +1,10 @@
 import { ComponentFactoryResolver, Directive, Input, OnDestroy, ViewContainerRef } from '@angular/core';
 import { IAlertComponent } from './alerts.model';
-import { IOverlay } from '@ansyn/imagery';
-
 
 export interface IAlertComponentSettings {
 	component: any
 	mapId: string;
-	overlay: IOverlay;
+	overlay: any;
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/alerts/alert-component.directive.ts]: TODO: to be injected elsewhere

File: src/app/@ansyn/map-facade/components/imagery-container/imagery-container.component.ts [hunk 8]
```diff
@@ -24,7 +24,7 @@ export class ImageryContainerComponent {
 		map((mapState: IMapState) => mapState.isHiddenMaps.has(this.mapState.id))
 	);
 
-	get overlay(): IOverlay {
+	get overlay(): any {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-container/imagery-container.component.ts]: Add TODO

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.html [hunk 9]
```diff
@@ -18,8 +18,9 @@
 		<i class="fa fa-arrows"></i>
 	</button>
 
-	<select *ngIf="false"
+	<select
 		class="combo-box-width"
+		hidden
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.html]: why?

File: src/app/@ansyn/map-facade/components/imagery-rotation/imagery-rotation.component.ts [hunk 10]
```diff
@@ -70,7 +74,7 @@ export class ImageryRotationComponent {
 	}
 
 	isGeoRegistered() {
-		return !this.mapState.data.overlay || (this.mapState.data.overlay.isGeoRegistered !== GeoRegisteration.notGeoRegistered);
+		return !this.mapState.data.overlay || (this.mapState.data.overlay.isGeoRegistered !== 'notGeoRegistered');
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-rotation/imagery-rotation.component.ts]: Add TODO

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts [hunk 11]
```diff
@@ -48,7 +49,7 @@ import { copyFromContent } from '../../utils/clipboard';
 	destroy: 'ngOnDestroy'
 })
 export class ImageryStatusComponent implements OnInit, OnDestroy {
-	_overlay: IOverlay;
+	_overlay: any;
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts]: Add TODO

File: src/app/@ansyn/map-facade/components/overlay-source-type-notice/overlay-source-type-notice.component.ts [hunk 12]
```diff
@@ -10,7 +9,7 @@ import { mapFacadeConfig } from '../../models/map-facade.config';
 })
 export class OverlaySourceTypeNoticeComponent {
 
-	@Input() set overlay(newOverlay: IOverlay) {
+	@Input() set overlay(newOverlay: any) {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/overlay-source-type-notice/overlay-source-type-notice.component.ts]: Add TODO

File: src/app/@ansyn/map-facade/reducers/imagery-status.reducer.ts [hunk 13]
```diff
@@ -98,8 +97,8 @@ export function ImageryStatusReducer(state: ImageryStatusState = imageryStatusIn
 
 export const selectRemovedOverlaysVisibility: MemoizedSelector<any, boolean> = createSelector(imageryStatusStateSelector, (imageryStatus) => imageryStatus.removedOverlaysVisibility);
 export const selectRemovedOverlaysIdsCount: MemoizedSelector<any, number> = createSelector(imageryStatusStateSelector, (imageryStatus) => imageryStatus.removedOverlaysIdsCount);
-export const selectFavoriteOverlays: MemoizedSelector<any, IOverlay[]> = createSelector(imageryStatusStateSelector, (imageryStatus) => imageryStatus.favoriteOverlays);
-export const selectPresetOverlays: MemoizedSelector<any, IOverlay[]> = createSelector(imageryStatusStateSelector, (imageryStatus) => imageryStatus.presetOverlays);
+export const selectFavoriteOverlays: MemoizedSelector<any, any[]> = createSelector(imageryStatusStateSelector, (imageryStatus) => imageryStatus.favoriteOverlays);
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/reducers/imagery-status.reducer.ts]: Add TODO

File: src/app/@ansyn/map-facade/reducers/map.reducer.ts [hunk 14]
```diff
@@ -37,9 +36,9 @@ export function setMapsDataChanges(oldEntities: Dictionary<any>, oldActiveMapId,
 	return mapsList;
 }
 
-export const mapsAdapter: EntityAdapter<ICaseMapState> = createEntityAdapter<ICaseMapState>();
+export const mapsAdapter: EntityAdapter<IMapSettings> = createEntityAdapter<IMapSettings>();
 
-export interface IMapState extends EntityState<ICaseMapState> {
+export interface IMapState extends EntityState<IMapSettings> {
 	activeMapId: string;
 	isLoadingMaps: Map<string, string>,
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/reducers/map.reducer.ts]: ; instead of ,

File: src/app/@ansyn/map-facade/services/map-facade.service.ts [hunk 15]
```diff
@@ -20,26 +14,26 @@ export class MapFacadeService {
...
-	mapsList: ICaseMapState[] = [];
+	mapsList: IMapSettings[] = [];
 
 	static isNotIntersect(extentPolygon, footprint, overlayCoverage): boolean {
 		const intersection = getFootprintIntersectionRatioInExtent(extentPolygon, footprint);
 		return intersection < overlayCoverage;
 	}
 
-	static isOverlayGeoRegistered(overlay: IOverlay): boolean {
+	static isOverlayGeoRegistered(overlay: any): boolean {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/services/map-facade.service.ts]: Add TODO

File: src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts [hunk 16]
```diff
@@ -1,12 +1,8 @@
 import { ToolsActions, ToolsActionsTypes } from '../actions/tools.actions';
 import { createFeatureSelector, createSelector, MemoizedSelector } from '@ngrx/store';
-import {
-	AnnotationMode,
-	ImageManualProcessArgs,
-	IOverlaysManualProcessArgs,
-	IVisualizerStyle,
-	OverlayDisplayMode
-} from '@ansyn/imagery';
+import { AnnotationMode, IVisualizerStyle } from '@ansyn/imagery';
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/modules/menu-items/tools/reducers/tools.reducer.ts]: IVisualizerStyle ?

File: src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts [hunk 17]
```diff
@@ -1,19 +1,19 @@
...
-} from '@ansyn/imagery';
 import {
 	IMarkUpData,
 	IOverlayDropMarkUp,
 	ITimelineRange,
 	MarkUpClass
 } from '../reducers/overlays.reducer';
 import { type } from '../../core/utils/type';
+import {
+	IOverlay,
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/modules/overlays/actions/overlays.actions.ts]: its not map facade

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts (hunk 11 above)
Comment (yairtawil) [reply src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts]: todo

File: src/app/@ansyn/map-facade/reducers/map.reducer.ts (hunk 14 above)
Comment (yairtawil) [reply src/app/@ansyn/map-facade/reducers/map.reducer.ts]: 👍

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts [hunk 18]
```diff
@@ -48,45 +50,61 @@ import { get as _get } from 'lodash';
...
+	mapsAmount = 1;
+	@HostBinding('class.active') isActiveMap: boolean;
+
+	get selectedMap() {
+		return _get(this.communicators.provide(this.mapState.id), 'ActiveMap.mapType');
+	}
+
+	@Input()
+	set mapState(value: IMapSettings) {
+		if (_get(this._mapState, 'data.overlay') !== _get(value, 'data.overlay')) {
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts]: i don't like the lodash way, can you find a better way?

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts [hunk 19]
```diff
@@ -253,23 +271,23 @@ export class ImageryStatusComponent implements OnInit, OnDestroy {
 	}
 
 	toggleMapLayers() {
-		this.layerFlag = !this.layerFlag;
-		this.store$.dispatch(new ToggleMapLayersAction({ mapId: this.mapId }));
+		// this.mapState.flags.displayLayers = !this.mapState.flags.displayLayers;
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts]: ???

File: src/app/@ansyn/map-facade/map-facade.module.ts [hunk 20]
```diff
@@ -60,7 +68,7 @@ import { AnsynPopoverComponent } from './components/ansyn-popover/ansyn-popover.
 		InfiniteScrollDirective,
 		AnimatedEllipsisComponent,
 		AnsynLoaderComponent,
-		AnsynPopoverComponent
+		AnsynPopoverComponent,
```
Comment (asafMasa) [inline src/app/@ansyn/map-facade/map-facade.module.ts]: remove ','

File: src/app/@ansyn/ansyn/components/overlay-out-of-bounds/overlay-out-of-bounds.component.ts [hunk 21]
```diff
@@ -1,21 +1,19 @@
...
-	@Input() mapId: string;
-	@Input() overlay: IOverlay;
+export class OverlayOutOfBoundsComponent implements IEntryComponent {
+	@Input() mapState: IMapSettings;
 
 	backToExtent(): void {
-		const communicator = this.communicatorService.provide(this.mapId);
-		const extent = extentFromGeojson(this.overlay.footprint);
+		const communicator = this.communicatorService.provide(this.mapState.id);
+		const extent = extentFromGeojson(this.mapState.data.overlay.footprint);
```
[the remark this answers is not in the document]
Comment (yairtawil) [reply src/app/@ansyn/ansyn/components/overlay-out-of-bounds/overlay-out-of-bounds.component.ts]: 👍

File: src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts (hunk 18 above)
Comment (yairtawil) [reply src/app/@ansyn/map-facade/components/imagery-status/imagery-status.component.ts]: lodash `get` is great 🥇

**Pull request merged**